### PR TITLE
Improve VisibleOnScreen classes' docs

### PR DESCRIPTION
**Pull request by MewPurPur**
These docs were super old and had some misleading information, unless I've misunderstood something. Some bad English too. And a few descriptions were missing.

I tested to reaffirm that I'm not getting things wrong, but I'm not a big user of these classes so I might have missed some peculiarity or something.

### CODE REVIEW

**Comment by Calinou** [comment]
cc @mrjustaguy, as they worked with this class previously.

File: doc/classes/VisibleOnScreenEnabler2D.xml [hunk 1]
```diff
@@ -1,19 +1,21 @@
...
-			Determines how the node is enabled. Corresponds to [enum Node.ProcessMode]. Disabled node uses [constant Node.PROCESS_MODE_DISABLED].
+			Determines how the target node is enabled. Corresponds to [enum Node.ProcessMode]. When the node is disabled, it always uses [constant Node.PROCESS_MODE_DISABLED].
 		</member>
 		<member name="enable_node_path" type="NodePath" setter="set_enable_node_path" getter="get_enable_node_path" default="NodePath(&quot;..&quot;)">
-			The path to the target node, relative to the [VisibleOnScreenEnabler2D]. The target node is cached; it's only assigned when setting this property (if the [VisibleOnScreenEnabler2D] is inside scene tree) and every time the [VisibleOnScreenEnabler2D] enters the scene tree. If the path is invalid, nothing will happen.
+			The path to the target node, relative to the [VisibleOnScreenEnabler2D]. The target node is cached; it's only assigned when setting this property (if the [VisibleOnScreenEnabler2D] is inside the scene tree) and every time the [VisibleOnScreenEnabler2D] enters the scene tree. If the path is invalid, nothing will happen.
```
Comment (mrjustaguy) [inline doc/classes/VisibleOnScreenEnabler2D.xml:18]: It'll Print an error in the editor, but in game it won't do anything.

File: doc/classes/VisibleOnScreenEnabler3D.xml [hunk 2]
```diff
@@ -1,28 +1,32 @@
...
-		[b]Note:[/b] VisibleOnScreenEnabler3D uses an approximate heuristic for performance reasons. It doesn't take walls and other occlusion into account. The heuristic is an implementation detail and may change in future versions. If you need precise visibility checking, use another method such as adding an [Area3D] node as a child of a [Camera3D] node and/or [method Vector3.dot].
-		[b]Note:[/b] VisibleOnScreenEnabler3D will not affect nodes added after scene initialization.
+		[VisibleOnScreenEnabler3D] contains a box-shaped region of 3D space and a target node. The target node will be automatically enabled (via its [member Node.process_mode] property) when any part of this region becomes visible on the screen, and automatically disabled otherwise. This can for example be used to activate enemies only when the player approaches them.
+		See [VisibleOnScreenNotifier3D] if you only want to be notified when the region is visible on screen.
+		[b]Note:[/b] [VisibleOnScreenEnabler3D] uses an approximate heuristic that doesn't take walls and other occlusion into account. This is an implementation detail and may change in future versions.
```
Comment (mrjustaguy) [inline doc/classes/VisibleOnScreenEnabler3D.xml:9]: This Note is somewhat wrong, it takes Occlusion into account if you are using Occlusion Culling and have set it up properly, it's using the same logic as the Notifier.

File: doc/classes/VisibleOnScreenEnabler3D.xml [hunk 3]
```diff
@@ -1,28 +1,32 @@
...
 	</description>
 	<tutorials>
 	</tutorials>
 	<members>
 		<member name="enable_mode" type="int" setter="set_enable_mode" getter="get_enable_mode" enum="VisibleOnScreenEnabler3D.EnableMode" default="0">
+			Determines how the target node is enabled. Corresponds to [enum Node.ProcessMode]. When the node is disabled, it always uses [constant Node.PROCESS_MODE_DISABLED].
 		</member>
 		<member name="enable_node_path" type="NodePath" setter="set_enable_node_path" getter="get_enable_node_path" default="NodePath(&quot;..&quot;)">
+			The path to the target node, relative to the [VisibleOnScreenEnabler3D]. The target node is cached; it's only assigned when setting this property (if the [VisibleOnScreenEnabler3D] is inside the scene tree) and every time the [VisibleOnScreenEnabler3D] enters the scene tree. If the path is invalid, nothing will happen.
```
Comment (mrjustaguy) [inline doc/classes/VisibleOnScreenEnabler3D.xml:18]: Like above, it'll print an error in the editor, but the game should keep on going fine

File: doc/classes/VisibleOnScreenNotifier3D.xml [hunk 4]
```diff
@@ -1,38 +1,38 @@
...
 	</brief_description>
 	<description>
-		The VisibleOnScreenNotifier3D detects when it is visible on the screen. It also notifies when its bounding rectangle enters or exits the screen or a [Camera3D]'s view.
-		If you want nodes to be disabled automatically when they exit the screen, use [VisibleOnScreenEnabler3D] instead.
-		[b]Note:[/b] VisibleOnScreenNotifier3D uses the render culling code to determine whether it's visible on screen, which also means that its [member Node3D.visible] must be [code]true[/code] to work correctly.
+		[VisibleOnScreenEnabler3D] represents a box-shaped region of 3D space. When any part of this region becomes visible on screen or in a [Camera3D]'s view, it will emit a [signal screen_entered] signal, and likewise it will emit a [signal screen_exited] signal when no part of it remains visible.
+		If you want a node to be disabled automatically when this region isn't visible on screen, use [VisibleOnScreenEnabler3D].
+		[b]Note:[/b] [VisibleOnScreenNotifier3D] uses the render culling code to determine whether it's visible on screen, so it won't function unless [member Node3D.visible] is set to [code]true[/code].
```
Comment (mrjustaguy) [inline doc/classes/VisibleOnScreenNotifier3D.xml:9]: emphasize that it works with Occlusion Culling

**Review by mrjustaguy — CHANGES_REQUESTED** [review]

**Review by mrjustaguy — APPROVED** [review]
Looks good to me

File: doc/classes/VisibleOnScreenNotifier2D.xml [hunk 5]
```diff
@@ -1,12 +1,12 @@
...
 	<brief_description>
-		Detects when the node extents are visible on screen.
+		A rectangular region of 2D space that detects whether it is visible on screen.
 	</brief_description>
 	<description>
-		The VisibleOnScreenNotifier2D detects when it is visible on the screen. It also notifies when its bounding rectangle enters or exits the screen or a viewport.
-		If you want nodes to be disabled automatically when they exit the screen, use [VisibleOnScreenEnabler2D] instead.
-		[b]Note:[/b] VisibleOnScreenNotifier2D uses the render culling code to determine whether it's visible on screen, which also means that its [member CanvasItem.visible] must be [code]true[/code] to work correctly.
+		[VisibleOnScreenEnabler2D] represents a rectangular region of 2D space. When any part of this region becomes visible on screen or in a viewport, it will emit a [signal screen_entered] signal, and likewise it will emit a [signal screen_exited] signal when no part of it remains visible.
+		If you want a node to be disabled automatically when this region isn't visible on screen, use [VisibleOnScreenEnabler2D].
```
Comment (YuriSizov) [inline doc/classes/VisibleOnScreenNotifier2D.xml:8]: Same here, I guess.

**Review by YuriSizov — APPROVED** [review]
Aside from the note that I have left, looks solid to me.

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**